ALICE3/TableProducer/OTF/onTheFlyTracker.cxx

```diff
@@ -299,41 +300,40 @@ struct OnTheFlyTracker {
   TRandom3 rand;
   Service<o2::ccdb::BasicCCDBManager> ccdb;
 
-  static constexpr int kMaxLUTConfigs = 20;
   void init(o2::framework::InitContext&)
   {
     ccdb->setURL("http://alice-ccdb.cern.ch");
     ccdb->setTimestamp(-1);
 
-    if (enablePrimarySmearing) {
-      auto loadLUT = [&](int icfg, int pdg, const std::vector<std::string>& tables) {
-        LOG(info) << "Loading LUT for pdg " << pdg << " for config " << icfg << " from provided tables with size " << tables.size();
-        if (tables.empty()) {
-          LOG(debug) << "No LUT file passed for pdg " << pdg << ", skipping.";
-          return false;
-        }
-        const bool foundNewCfg = static_cast<size_t>(icfg) < tables.size();
-        std::string lutFile = foundNewCfg ? tables[icfg] : tables.front();
-        LOG(info) << "Loading LUT for pdg " << pdg << " from file " << lutFile << " for config " << icfg;
-        // strip from leading/trailing spaces
-        lutFile.erase(0, lutFile.find_first_not_of(" "));
-        lutFile.erase(lutFile.find_last_not_of(" ") + 1);
-        if (lutFile.empty()) {
-          LOG(debug) << "Empty LUT file name for pdg " << pdg << ", skipping.";
-          return false;
-        }
-        bool success = mSmearer[icfg]->loadTable(pdg, lutFile.c_str());
-        if (!success) {
-          LOG(fatal) << "Having issue with loading the LUT " << pdg << " " << lutFile;
-        }
-
-        return foundNewCfg;
-      };
+    auto loadLUT = [&](int icfg, int pdg, const std::vector<std::string>& tables) {
+      LOG(info) << "Loading LUT for pdg " << pdg << " for config " << icfg << " from provided tables with size " << tables.size();
+      if (tables.empty()) {
+        LOG(debug) << "No LUT file passed for pdg " << pdg << ", skipping.";
+        return false;
+      }
+      const bool foundNewCfg = static_cast<size_t>(icfg) < tables.size();
+      std::string lutFile = foundNewCfg ? tables[icfg] : tables.front();
+      LOG(info) << "Loading LUT for pdg " << pdg << " from file " << lutFile << " for config " << icfg;
+      // strip from leading/trailing spaces
+      lutFile.erase(0, lutFile.find_first_not_of(" "));
+      lutFile.erase(lutFile.find_last_not_of(" ") + 1);
+      if (lutFile.empty()) {
+        LOG(info) << "Empty LUT file name for pdg " << pdg << ", skipping.";
+        return false;
+      }
+      bool success = mSmearer[icfg]->loadTable(pdg, lutFile.c_str());
+      if (!success) {
+        LOG(fatal) << "Having issue with loading the LUT " << pdg << " " << lutFile;
+      }
 
-      for (int icfg = 0; icfg < kMaxLUTConfigs; ++icfg) {
-        mSmearer.emplace_back(std::make_unique<o2::delphes::DelphesO2TrackSmearer>());
-        mSmearer[icfg]->setCcdbManager(ccdb.operator->());
+      return foundNewCfg;
+    };
 
+    for (int icfg = 0; icfg < static_cast<int>(fastTrackerSettings.alice3geo->size()); ++icfg) {
```

Comment:
Hi @lhusova! I think that this could be problematic for analyses that does not use the fastTracker such as the D0 and Lc. Maybe we could instead use the size of the lut vectors for the primary particles?

Reply:
Hi @lhusova, I think that doing it as two loops would be OK. One just needs to be careful such that the index matches between the loop for the primaries and the secondaries.
Because in my analysis for example, I use both, so it's important for me that one index points to the same geometry in both cases so I don't use one geometry for primaries and a different one for secondaries

The process function would also have to be adjusted similarly then I think [onTheFlyTracker.cxx#L1544](https://github.com/AliceO2Group/O2Physics/blob/master/ALICE3/TableProducer/OTF/onTheFlyTracker.cxx#L1544 )

Reply:
Hi @jesgum, please have a look how I solved it. I don't think that the line 1544 needs a change as the smearer is created as many times as this loop.